src/plugins/intel_gpu/tests/test_cases/experimental_detectron_generate_proposals_single_image_gpu_test.cpp

```diff
@@ -151,12 +246,20 @@ struct experimental_detectron_generate_proposals_single_image_test
 
         const auto outputs = network.execute();
 
-        const auto rois = outputs.at(edgpsi_id).get_memory();
+        const auto rois = outputs.at(reorder_result_id).get_memory();
 
         const cldnn::mem_lock<T> rois_ptr(rois, get_test_stream());
         ASSERT_EQ(rois_ptr.size(), param.post_nms_count * 4);
 
-        const cldnn::mem_lock<T> roi_scores_ptr(output_roi_scores, get_test_stream());
+        cldnn::topology reorder_topology;
+        reorder_topology.add(input_layout("scores", rois_scores_layout));
```

Comment:
can't all those transformations be done by the previous network, mean by network.execute()

Reply:
No. Plugin does not support multiple outputs natively, and scores and rois_num are, technically, inputs of the primitive.
From network prospective, they are not dependent on GenerateProposals layer, so, due to optimization their reorders would happen in the very beginning, before GenerateProposals. That's why we need separate network to reorder additional outputs.